dependencies-in-practice.Rmd

```diff
@@ -678,35 +678,24 @@ export(count)
 
 Now imagine that your package implements a method for `count()` for a class you "own" (not `data.frame`).
 A good example is the dbplyr package, which implements `count()` for the `tbl_lazy` class.
-An add-on package that implements an S3 generic for a new class should list the generic-providing package in `Imports`, import the generic into its namespace, and export its S3 method.
-Here's part of dbplyr's `DESCRIPTION` file:
-
-```         
-Imports: 
-    ...,
-    dplyr,
-    ...
-```
 
-In `dbplyr/R/verb-count.R`, we have:
+In this case, `@export` will not work because it assumes the `count()` generic is included or imported in the dbplyr NAMESPACE, so instead we need to use `@exportS3Method`.
+This will use "conditional" method registration, which means the method will only be registered when the dplyr package is loaded.
 
 ```{r}
 #| eval: false
-#' @importFrom dplyr count
-#' @export
+#' @exportS3Method dplyr::count
 count.tbl_lazy <- function(x, ..., wt = NULL, sort = FALSE, name = NULL) { ... }
 ```
 
 In `NAMESPACE`, we have:
 
 ```         
 S3method(count,tbl_lazy)
-...
-importFrom(dplyr,count)
 ```
 
-Dbplyr also provides methods for various generics provided by the base package, such as `dim()` and `names()`.
-In this case, there is no need to import those generics, but it's still necessary to export the methods.
+dbplyr also provides methods for various generics provided by the base package, such as `dim()` and `names()` - in this case, we can use `@export` or `@exportS3Method base::names`
```

Comment:
I think I'd still recommend using `@export` here because we don't need conditional registration.

Reply:
```suggestion
dbplyr also provides methods for various generics provided by the base package, such as `dim()` and `names()` - in this case, we can still use `@export` since we don't need conditional registration. 
```